components/net/resource_thread.rs

```diff
@@ -317,7 +321,9 @@ impl CoreResourceManager {
     }
 
     fn fetch(&self,
-             init: RequestInit,
+             req_init: RequestInit,
+             res_init: ResponseInit,
+             redirect_flag: bool,
```

Comment:
Instead of two arguments, we can have a single `redirected_from: Option<ResponseInit>`.